Cogs/ClassManagement.py

```diff
@@ -0,0 +1,256 @@
+
+        Outputs:
+            categories (List[categories]): a list of the verified categories.
+        
+        """
+
+        # Get list of all categories (category objects) to be destroyed and print
+        categories = []
+        missing_categories = '__**MISSING FOLLOWING CATEGORIES**__\n'
+
+        for category_name in category_names:
+            category = get(ctx.guild.categories, name=category_name)
+
+            # If the category was not found it adds it to the missing_categories message
+            if category == None:
+                missing_categories += f'{category_name}\n'
+            else:
+                categories.append(category)
+
+        # only send if there are missing categories
+        if not len(categories) == len(category_names):
+            await ctx.send(missing_categories)
+
+        return categories
+
+    async def get_roles(self, ctx, role_names):
+        """ Looks through all roles and verifies if the list of role names is on the server
+
+        Args:
+            role_names (List[str]): a list of strings of the role names to delete
+
+        Outputs:
+            roles (List[roles]): a list of the verified roles.
+
+        """
+
+        # Get list of all categories (category objects) to be destroyed and print
+        roles = []
+        missing_roles = '__**MISSING FOLLOWING ROLES**__\n'
+
+        for role_name in role_names:
+            role = get(ctx.guild.roles, name=role_name)
+
+            # If the role was not found it adds it to the missing_roles message
+            if role == None:
+                missing_roles += f'{role_name}\n'
+            else:
+                roles.append(role)
+        
+        # only send if there are missing roles
+        if not len(roles) == len(role_names):
+            await ctx.send(missing_roles)
+
+        return roles
+
+    @commands.command()
+    @commands.has_permissions(administrator=True)
+    async def buildcourses(self, ctx):
+        """Create class channels
+
+        Try to destroy old class channels (calls on destroycourses)
+        Read in role csv from cached file or from attachment on discord message
+        Get confirmation from author
+        Create all categories, channels, and roles
+        Call command to create role menus
+        """
+
+        #destroy classes first here
+        await self.destroycourses(ctx)
+
+        # Load roles csv
+        csv_filepath = f'role_lists/roles_{ctx.guild.id}.csv'
+
+        # If csv file attached, overwrite existing csv
+        if len(ctx.message.attachments) > 0:
+            try:
+                os.remove(csv_filepath)
+            except FileNotFoundError:
+                pass
+            await ctx.message.attachments[0].save(csv_filepath)
+        
+        # Load roles csv
+        courses_df = pd.read_csv(csv_filepath)
+
+        courses_df = courses_df.dropna(subset=['create_channels'])
+
+        role_names=courses_df["role/link"].to_list()
```

Comment:
I think the "proper" way to iterate like this is to use https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.itertuples.html

I'm not entirely sure, but it will probably look a bit cleaner.

Reply:
#151